docs/adr/0021-collab-mvp-schema-design.md

```diff
@@ -0,0 +1,118 @@
+
+## Status
+
+Proposed
+
+## Context
+
+[TTAHUB-3473](https://jira.acf.gov/browse/TTAHUB-3473) is tracking the work to create an MVP (Minium Viable Product) for Collaboration (collab) Reports. Collab reports are a new entity within the TTA Hub and as such new database constructs ( tables, indices etc.) are needed. This ADR will document a proposed schema to serve as basis for the database design, prior to implementation.
+
+## Decision
+
+The following is proposed design of the tables required to support the collab report MVP. Non-Primary key indices have not been included as it is not yet clear if any are needed ( can be added during the MVP process as needed). Data types and and attributes have been expressed as PostgreSQL provided constructs, rather than the abstractions provide by [Sequelize](https://sequelize.org/docs/v7/models/data-types/)
+
+
+### Terminology
+
+The following are a list of mnemonics and abbreviations used through the rest of the section:
+
+- AUTO - Auto Generated/SERIAL Value
+- FK - Foreign Key
+- PK - Primary Key
+
+### Overall Design Comments
+
+- The `enum` type has been used heavily given that there are many collab reports attributes that are presented to the user as a list of options rather than arbitrary strings. These could also be represented as dictionary-style tables in their own right if desired.
+- "Supporting tables" ( i.e. those that exist only to represent a non-scalar attribute of a single collab report ) have been designed without any auto-generated PK column, instead using a multiple column PK (of which, the associated collab report ID is always included). This enforces inherent uniqueness of the additional primary key values without additional attributes / logic.
+
+### Tables
+
+#### CollabReport
+
+Purpose: This is the top-level "root" table for Collaboration Reports. Scalar properties of collab reports ( that are not calculated at run/read time ) should be stored as columns here. Non-scalar properties are stored in other tables and that all share include the `reportId` column in this table as a primary key.
+
+| ColumnName | DataType | Required? | Allow Null? |  UI DataType | Attributes | Notes |
+|------------|----------|-----------|-------------|--------------|------------|-------|
+| reportId | INTEGER | Yes | No | none (generated) | PK, AUTO | |
+| name | VARCHAR | Yes | No | String |||
+| status | ENUM | Yes | No | None/String || `[DRAFT,SUBMITTED,REVIEWED,NEEDS_APPROVAL,APPROVED]` |
+| startDate | DATE | Yes | No | YYYYMMDD |||
+| endDate | DATE | Yes | No | YYYYMMDD |||
+| duration | SMALLINT | Yes | No | non-negative integer |||
+| isStateActivity | boolean | Yes | No | Binary Radio Buttons || If there will EVER be a third option to regional/state then this should be a different type, but assume this is unlikely |
+| conductMethod | enum | Yes | No | CHOICE ||`[EMAIL, PHONE, IN_PERSON, VIRTUAL]`|
+| description | TEXT | Yes | No | Text Area |||
```

Comment:
I am assuming that the `createdAt` and `updatedAt` are not in any of the tables described here because they're Sequelize-managed default fields, but I think we want to add `deletedAt` for `paranoid` use to this base-level table. That pattern is missing in `ActivityReports` because it was an early table and we made "deleted" a status ( along with e.g. "submitted" and "approved"), but `deletedAt` is a nicer and more standard way to do it.